nav-app/src/app/app.component.ts

```diff
@@ -1,36 +1,59 @@
-import { Component, ViewChild, DoCheck, HostListener } from '@angular/core';
+import { Component, ViewChild, HostListener, OnInit } from '@angular/core';
 import { TabsComponent } from './tabs/tabs.component';
 import { ConfigService } from './config.service';
 import * as globals from "./globals";
+import { IconsService } from "./icons.service";
+import { deleteCookie, getCookie, hasCookie, setCookie } from "./cookies";
 
 @Component({
-    selector: 'app-root',
-    templateUrl: './app.component.html',
-    styleUrls: ['./app.component.scss'],
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss'],
 })
-export class AppComponent {
-    @ViewChild(TabsComponent) tabsComponent;
+export class AppComponent implements OnInit {
+  @ViewChild(TabsComponent) tabsComponent;
 
-    nav_version: string = globals.nav_version;
-    quantumCustomizationVersion: string = globals.QUANTUM_CUSTOMIZATION_VERSION;
+  nav_version: string = globals.nav_version;
+  quantumCustomizationVersion: string = globals.QUANTUM_CUSTOMIZATION_VERSION;
+  public user_theme: string;
 
-    // @HostListener('window:beforeunload', ['$event'])
-    // promptNavAway($event) {
-    //     if (!this.configService.getFeature('leave_site_dialog')) return;
-    //     //this text only shows in the data, not visible to user as far as I can tell
-    //     //however, if it's not included the window doesn't open.
-    //     $event.returnValue='Are you sure you want to navigate away? Your data may be lost!';
-    // }
+  @HostListener('window:beforeunload', ['$event'])
+  promptNavAway($event) {
+    if (!this.configService.getFeature('leave_site_dialog')) return;
+    //this text only shows in the data, not visible to user as far as I can tell
+    //however, if it's not included the window doesn't open.
+    $event.returnValue = 'Are you sure you want to navigate away? Your data may be lost!';
+  }
```

Comment:
Did we previously comment this out?

Reply:
Yes it was originally commented but I was not really sure why it was commented

Reply:
I think I have an idea why we commented this out :) anyway I've commented this because the `Leave dialogue` is quite nagging

Reply:
Maybe we should leave this uncommented and set it to `false` in  https://github.com/quantum-sec/attack-navigator/blob/d643eb292b478fe2703e9b7f87fec4bdb658685d/nav-app/src/assets/config.json#L123